src/core_plugins/i18n/server/i18n/i18n.js

```diff
@@ -0,0 +1,130 @@
+import path from 'path';
+import Promise from 'bluebird';
+import { readFile } from 'fs';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+let registeredTranslations = {};
+
+/**
+ * Return all translations registered for a particular locale.
+ * @param {string} locale - Translation locale to be returned
+ * @return {Promise} - A Promise object which will contain on resolve a JSON object of all registered translations
+ */
+export function getTranslationsForLocale(locale) {
```

Comment:
Yeah, agree with your synopsis. The more we iterate we begin to see it simplify, I hope! :-)

I have changed it around where there is just one exported function now called 'getTranslations(languageTags)'. This can be called with 1..m tags and will do exact or best case match and tghen get the translations for the locale returned. 

Let me know what you think.